### ISSUE TITLE
[BUG] Docker mods broken

### ISSUE THREAD

**Issue by tomislav** [linuxserver/docker-sonarr#275]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Ever since cc03eefbfc57c7f71cabd2139c08754a694e7712 docker mods seem to be broken. Startup hangs at 

```
[mod-init] Attempting to run Docker Modification Logic
```

The env value is

DOCKER_MODS=mdhiggins/sma-mod:latest

If I remove the env value, the container starts up fine

### Expected Behavior

Docker mod should load and sonarr should start up

### Steps To Reproduce

1. In the "latest" tag, add DOCKER_MODS=mdhiggins/sma-mod:latest
2. Start sonarr

### Environment

```markdown
- OS: DSM 7.1.1-42962 Update 6
- How docker service was installed: System docker
```

### CPU architecture

x86-64

### Docker creation

```bash
{
   "CapAdd" : [],
   "CapDrop" : [],
   "cmd" : "",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : true,
   "enabled" : false,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/lsiopy/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "PS1",
         "value" : "$(whoami)@$(hostname):$(pwd)\\$ "
      },
      {
         "key" : "HOME",
         "value" : "/root"
      },
      {
         "key" : "TERM",
         "value" : "xterm"
      },
      {
         "key" : "S6_CMD_WAIT_FOR_SERVICES_MAXTIME",
         "value" : "0"
      },
      {
         "key" : "S6_VERBOSITY",
         "value" : "1"
      },
      {
         "key" : "S6_STAGE2_HOOK",
         "value" : "/docker-mods"
      },
      {
         "key" : "VIRTUAL_ENV",
         "value" : "/lsiopy"
      },
      {
         "key" : "LSIO_FIRST_PARTY",
         "value" : "true"
      },
      {
         "key" : "XDG_CONFIG_HOME",
         "value" : "/config/xdg"
      },
      {
         "key" : "SONARR_BRANCH",
         "value" : "phantom-develop"
      },
      {
         "key" : "LANGUAGE",
         "value" : "en_US.UTF-8"
      },
      {
         "key" : "LANG",
         "value" : "en_US.UTF-8"
      },
      {
         "key" : "PUID",
         "value" : "1024"
      },
      {
         "key" : "PGID",
         "value" : "65536"
      },
      {
         "key" : "TZ",
         "value" : "Europe/Zagreb"
      },
      {
         "key" : "DOCKER_MODS",
         "value" : "mdhiggins/sma-mod:latest"
      },
      {
         "key" : "SMA_UPDATE",
         "value" : "true"
      }
   ],
   "exporting" : false,
   "id" : "d8c0fc5251cbaa772765af50105f0fa8ae4026000a6acfcf1ba8e18cd3cb796c",
   "image" : "linuxserver/sonarr:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "sonarr",
   "network" : [
      {
         "driver" : "host",
         "name" : "host"
      }
   ],
   "network_mode" : "host",
   "port_bindings" : [],
   "privileged" : false,
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : true,
   "volume_bindings" : [
      {
         "host_volume_file" : "/Storage",
         "mount_point" : "/data",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/sonarr",
         "mount_point" : "/config",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/sma",
         "mount_point" : "/usr/local/sma/config",
         "type" : "rw"
      }
   ]
}
```

### Container logs

```bash
[mod-init] Attempting to run Docker Modification Logic
```

**Comment by j0nnymoe** [comment]
You would need to report this to the mod developer.

**Comment by tomislav** [comment]
@j0nnymoe Everything worked fine when using the "develop" tag (v4) before it this container was updated to "stable" in https://github.com/linuxserver/docker-sonarr/commit/cc03eefbfc57c7f71cabd2139c08754a694e7712

**Comment by thespad** [comment]
I suspect you're running into https://info.linuxserver.io/issues/2023-12-30-synology/ as the first steps of the mod logic involve a series of calls with curl to the docker registry

**Comment by tomislav** [comment]
@thespad yes, I can see now that a curl request to docker-mods blacklist.txt is stuck

**Comment by thespad** [comment]
It's not something we can directly fix, for now you'll have to roll back to the older develop tag (or run Sonarr on another docker host with a non-EOL kernel).

**Comment by tomislav** [comment]
Just for reference if anyone is having the same issue.

This is a [bug in the c-ares DNS library](https://github.com/c-ares/c-ares/issues/660) that has been fixed in [version 1.24.0](https://c-ares.org/changelog.html#1_24_0), and that version has been [commited to the Alpine Linux packages index](https://github.com/alpinelinux/aports/blob/master/main/c-ares/APKBUILD) a week ago. 

I'm not sure how packages are updated in Alpine Linux, but hopefully it will be fixed soon.

**Comment by thespad** [comment]
The master branch is Edge which is their rolling release channel and those changes won't normally be backported to older releases.

We have requested it for 3.19, but I can't give you any sense of if or when it might get merged.
